[RISCV] Add Supm extension to RVA23 profiles

asb opened the pull request:
This is mandatory for both RVA23U64 and RVA23S64 in the ratified version of the specification
<https://github.com/riscv/riscv-profiles/blob/main/src/rva23-profile.adoc>.

wangpc-pp reviewed (approved):
LGTM.
(And it's time to mark RV[A|B|M]23 as non-experimental now?)

asb commented:
> LGTM. (And it's time to mark RV[A|B|M]23 as non-experimental now?)

RVA23 and RVB23 yes. RVM23 isn't ratified yet. I've got a couple of other patches like this to lay the groundwork before flipping the switch.

asb committed:
[RISCV] Add Supm extension to RVA23 profiles

This is mandatory for both RVA23U64 and RVA23S64 in the ratified version
of the specification
<https://github.com/riscv/riscv-profiles/blob/main/src/rva23-profile.adoc>.
